### ISSUE TITLE
Indent bug for String that can't be disabled

### ISSUE THREAD

**Issue by vRallev** [pinterest/ktlint#1155]
Given:
```kotlin
fun test() {
  FunSpec.builder("injectMembers")
    .addModifiers(OVERRIDE)
    .addParameter("instance", classType)
    .apply {
      parameters.forEachIndexed { index, parameter ->
        val property = injectProperties[index]
        val propertyName = property.nameAsSafeName.asString()

        addStatement(
          /* ktlint-disable indent */
          "inject${propertyName.capitalize(US)}(instance, ${
            when {
              parameter.isWrappedInProvider -> parameter.name
              parameter.isWrappedInLazy ->
                "$daggerDoubleCheckFqNameString.lazy(${parameter.name})"
              else -> parameter.name + ".get()"
            }
          })"
          /* ktlint-enable indent */
        )
      }
    }
}
```
I'd expect that KtLint wouldn't complain about the indent for the `when` within the String, because that's how IntelliJ wants to format the code. I tried to disable the indent rule as you can see for this piece of code, but KtLint keeps complaining:
```
/Users/ralf/Development/projects/square/anvil/compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/MembersInjectorGenerator.kt:12:1 Unexpected indentation (12) (should be 10) (indent)
/Users/ralf/Development/projects/square/anvil/compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/MembersInjectorGenerator.kt:13:1 Unexpected indentation (14) (should be 12) (indent)
/Users/ralf/Development/projects/square/anvil/compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/MembersInjectorGenerator.kt:14:1 Unexpected indentation (14) (should be 12) (indent)
/Users/ralf/Development/projects/square/anvil/compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/MembersInjectorGenerator.kt:15:1 Unexpected indentation (16) (should be 14) (indent)
/Users/ralf/Development/projects/square/anvil/compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/MembersInjectorGenerator.kt:16:1 Unexpected indentation (14) (should be 12) (indent)
/Users/ralf/Development/projects/square/anvil/compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/MembersInjectorGenerator.kt:17:1 Unexpected indentation (12) (should be 10) (indent)
```
You should be able to reproduce the issue with `0.41.0` and the snippet as top level function. It's not necessary to resolve all imports.

**Comment by paul-dingemans** [comment]
This problem is not related to disabling indentation of a string. Currently, the indentation rule can only be disabled for an entire file and not for specific parts of the file. See issue https://github.com/pinterest/ktlint/issues/967 in which the generic issue is described.

**Comment by romtsn** [comment]
Yeah it's a dupe of #967
